sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -1344,7 +1350,7 @@ class CudfRegexTranspiler(mode: RegexMode) {
             // NOTE: (\A)* can be transpiled to (\A)?
             // however, (\A)? is not supported in libcudf yet
             case _ =>
-              val unsupportedTerm = isSupportedRepetitionBase(term)._2.get
+              val unsupportedTerm = getUnsupportedRepetitionBase(term).get
               throw new RegexUnsupportedException(
                 s"cuDF does not support repetition of group containing: " +
                   s"${unsupportedTerm.toRegexString}", term.position)
```

Comment:
```suggestion
              getUnsupportedRepetitionBase(term) match {
                 case Some(unsupportedTerm) =>
                    throw new RegexUnsupportedException(
                      s"cuDF does not support repetition of group containing: " +
                        s"${unsupportedTerm.toRegexString}", term.position)
                  case _ =>
               }
```

You can use a pattern match here to clean this up.

Reply:
There are a few more examples in the code as well.

Reply:
Then what I would suggest is not to return an Option from `getUnsupportedRepetitionBase(term)`.  In this case throw this kind of exception in that method itself. Basically if that method cannot return the `RegexAST` you want, you should throw a more meaningful exception there.